operations/mimir/read-write-deployment.libsonnet

```diff
@@ -0,0 +1,452 @@
+      'store-gateway.sharding-ring.zone-awareness-enabled': true,
+
+      // Use a different prefix so that both single-zone and multi-zone store-gateway rings can co-exists.
+      'store-gateway.sharding-ring.prefix': 'multi-zone/',
+
+      // Do not unregister from ring at shutdown, so that no blocks re-shuffling occurs during rollouts.
+      'store-gateway.sharding-ring.unregister-on-shutdown': false,
+    })) +
+    container.withVolumeMountsMixin([volumeMount.new('mimir-backend-data', '/data')]) +
+    $.util.resourcesRequests(1, '12Gi') +
+    $.util.resourcesLimits(null, '18Gi') +
+    $.util.readinessProbe +
+    $.jaeger_mixin,
+
+  newMimirBackendZoneStatefulset(zone, container)::
+    local name = 'mimir-backend-zone-%s' % zone;
+
+    $.newMimirStatefulSet(name, 3, container, mimir_backend_data_pvc) +
+    statefulSet.mixin.metadata.withLabels({ 'rollout-group': 'mimir-backend' }) +
+    statefulSet.mixin.metadata.withAnnotations({ 'rollout-max-unavailable': std.toString($._config.mimir_backend_max_unavailable) }) +
+    statefulSet.mixin.spec.template.metadata.withLabels({ name: name, 'rollout-group': 'mimir-backend' }) +
+    statefulSet.mixin.spec.selector.withMatchLabels({ name: name, 'rollout-group': 'mimir-backend' }) +
+    statefulSet.mixin.spec.updateStrategy.withType('OnDelete') +
+    statefulSet.mixin.spec.withReplicas(std.ceil($._config.mimir_backend_replicas / 3)) +
+    statefulSet.mixin.spec.template.spec.withTerminationGracePeriodSeconds(900) +
+    $.mimirVolumeMounts +
+    (if $._config.memberlist_ring_enabled then gossipLabel else {}) +
+    (if $._config.mimir_backend_allow_multiple_replicas_on_same_node then {} else {
+       spec+:
+         // Allow to schedule 2+ mimir-backends in the same zone on the same node, but do not schedule 2+ mimir-backends in
+         // different zones on the same node. In case of 1 node failure in the Kubernetes cluster, only mimir-backends
+         // in 1 zone will be affected.
+         podAntiAffinity.withRequiredDuringSchedulingIgnoredDuringExecution([
+           podAntiAffinity.requiredDuringSchedulingIgnoredDuringExecutionType.new() +
+           podAntiAffinity.requiredDuringSchedulingIgnoredDuringExecutionType.mixin.labelSelector.withMatchExpressions([
+             { key: 'rollout-group', operator: 'In', values: ['mimir-backend'] },
+             { key: 'name', operator: 'NotIn', values: [name] },
+           ]) +
+           podAntiAffinity.requiredDuringSchedulingIgnoredDuringExecutionType.withTopologyKey('kubernetes.io/hostname'),
+         ]).spec,
+     }),
+
+  // Creates a headless service for the per-zone mimir-backends StatefulSet. We don't use it
+  // but we need to create it anyway because it's responsible for the network identity of
+  // the StatefulSet pods. For more information, see:
+  // https://v1-18.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#statefulset-v1-apps
```

Comment:
Nit: this link doesn't seem to work anymore.

Reply:
Fixed.